### ISSUE TITLE
BC35000	Requested operation is not available because the runtime library function 'Microsoft.VisualBasic.CompilerServices.Conversions.ChangeType' is not defined.

### ISSUE THREAD

**Issue by twenzel** [dotnet/roslyn#28021]
We're trying to migrate VB.NET projects to new sdk project format. Following error occurs for some legacy code:
`BC35000	Requested operation is not available because the runtime library function 'Microsoft.VisualBasic.CompilerServices.Conversions.ChangeType' is not defined.`

Steps to reproduce:
ClassLibraryVBTest.vbproj:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <RootNamespace>ClassLibraryVBTest</RootNamespace>
    <TargetFramework>net472</TargetFramework>   
  </PropertyGroup>
  <PropertyGroup>
    <DefineConstants>NETFRAMEWORK,NET472,</DefineConstants>
    <OptionStrict>On</OptionStrict>
  </PropertyGroup>
  <ItemGroup>
    <Reference Include="Microsoft.VisualBasic" />
  </ItemGroup>
</Project>
```

Class1.vb
```
Option Strict Off
Public Class Class1
    Public Sub Test()
        Dim myObject As Object = CreateObject()


        Dim filename As String = ""
        myObject.Attachment.Add(filename) ' <--  error occurs here
    End Sub

    Private Function CreateObject() As Object
        Throw New NotImplementedException()
    End Function
End Class
```

Actual result:
error BC35000

Expected result:
No compiler error.

I'm using VS 2017 15.7.4.

**Comment by derkyjadex** [comment]
This seems to be related to embedding of the VB runtime. The `vbc.exe` command generated by the old style projects doesn't mention anything about the VB runtime, which is equivalent to running it with `/vbruntime+` switch (https://docs.microsoft.com/en-us/dotnet/visual-basic/reference/command-line-compiler/vbruntime).

The `vbc.exe` command generated by the new SDK style projects sticks `/vbruntime*` in there for some reason, which means the library should be embeded. This embedded version is apparently missing a bunch of features, including `Conversions.ChangeType`. This flag is controlled by the MSBuild variable `VBRuntime`, which defaults to `"Embed"`. According to [the source](http://source.roslyn.io/#Microsoft.Build.Tasks.CodeAnalysis/Vbc.cs,440), using the value `"Default"` will set the flag to the same as it effectively was before.

TL;DR: stick `<VBRuntime>Default</VBRuntime>` at the top of your `.vbproj` to restore the old behaviour.

I can't find anything to do with VB runtime embedding in the Visual Studio UI and it seems a little odd that the default behaviour is not to use the default setting. Although the fix is simple enough, it's not particularly easy to find.

**Comment by twenzel** [comment]
Thanks a lot, that helped.
But as @derkyjadex mentioned, there should be an UI to define this behaviour. Or, at least, a more prominent note in the documentation about this.

**Comment by jaredpar** [comment]
Closing as this is behaving as designed here. The project file format did indeed change the defaults around embedding the runtime.

> But as @derkyjadex mentioned, there should be an UI to define this behaviour. Or, at least, a more prominent note in the documentation about this.

Agree it would be nice if this was documented by the SDK team (CC @dsplaisted) as a deviation.

**Comment by MrM40** [comment]
The solution in my case was to insert `<VBRuntime>Default</VBRuntime>` in the `<PropertyGroup>` element, not in the root of the `.vbproj` file.
